config/webpack.config.api.js

```diff
@@ -0,0 +1,74 @@
+// @flow
+
+// Do this as the first thing so that any code reading it knows the right env.
+process.env.NODE_ENV = process.env.NODE_ENV || "development";
+process.env.BABEL_ENV = process.env.NODE_ENV;
+
+const webpack = require("webpack");
+const ModuleScopePlugin = require("react-dev-utils/ModuleScopePlugin");
+const RemoveBuildDirectoryPlugin = require("./RemoveBuildDirectoryPlugin");
+const paths = require("./paths");
+const getClientEnvironment = require("./env");
+
+const env = getClientEnvironment(null);
+
+module.exports = {
+  // Don't attempt to continue if there are any errors.
+  bail: true,
+  node: {
+    // Don't munge `__dirname` and `__filename`.
+    // https://github.com/webpack/webpack/issues/1599#issuecomment-186841345
+    __dirname: false,
+    __filename: false,
+  },
+  entry: {
+    api: paths.apiIndexJs,
+  },
+  output: {
+    path: paths.apiBuild,
+    // Generated JS file names (with nested folders).
+    // There will be one main bundle, and one file per asynchronous chunk.
+    // We don't currently advertise code splitting but Webpack supports it.
+    filename: "[name].js",
+    chunkFilename: "[name].[chunkhash:8].chunk.js",
+    libraryTarget: "umd",
+    // Use `this` for compatibility with both Node and browser.
+    globalObject: "this",
+  },
+  resolve: {
+    extensions: [".js", ".json"],
+    plugins: [
+      // Prevents users from importing files from outside of src/ (or node_modules/).
+      // This often causes confusion because we only process files within src/ with babel.
+      // To fix this, we prevent you from importing files out of src/ -- if you'd like to,
+      // please link the files into your node_modules/ and let module-resolution kick in.
+      // Make sure your source files are compiled, as they will not be processed in any way.
+      new ModuleScopePlugin(paths.appSrc, [paths.appPackageJson]),
+    ],
+  },
+  module: {
+    strictExportPresence: true,
+    rules: [
+      {
+        // "oneOf" will traverse all following loaders until one will
+        // match the requirements. If no loader matches, it will fail.
+        oneOf: [
+          // Process JS with Babel.
+          {
+            test: /\.(js|jsx|mjs)$/,
```

Comment:
If you wish to review this code, you may find it simplest to look at the
diff from the forked version:

```
diff -u config/webpack.config.backend.js config/webpack.config.api.js
```

This loader is uniform across all our loaders, which I think is good for
the principle of least surprise. It would be surprising if a utility
file should compilation errors in one build but not the others because
it’s being passed through Babel only in some configs.

That said, this exact kind of thing is why (at my suggestion) this
repository uses `.js` extensions for all files. Whether a file “should”
be `.js` or `.jsx` is often unclear, and the distinction doesn’t really
provide any value. So all of our stuff is `.js`, and this loader test is
just copied all the way back from the standard configs.

Reply:
> which I think is good for the principle of least surprise.

Makes sense :]

> Whether a file “should” be .js or .jsx is often unclear,

Yeah the double edged sword of JSX. The functional components style with state reduction has been my favorite way to separate concerns.

Anyway, since this is less surprising for the current setup, removing JSX support would also be a thing for when the frontend has moved it's own module.